### PR TITLE
update gradle build configuration

### LINKED ISSUE #13593: Remove/Cleanup Gradle Plugins
##### **Overview of the feature request**

We use a lot of plugins, some are which there for "historically" reasons.

I propose e.g. to use the apt plugins as they are not used/have any meaningful effect. 
This might effect our eclipse users, but I guess we can solve this by better documentation.
From my experience at least cli and idea work very well without apt plugins (as expected)

See also 

* https://github.com/tbroyer/gradle-apt-plugin
* https://plugins.gradle.org/plugin/com.diffplug.eclipse.apt
 
##### **Motivation for or Use Case**

We would like to show best practice and using such plugins contradicts this purpose and may also increase build times which is bad.

##### **Related issues or PR**

None

-   [x] Checking this box is mandatory (this is just to show you read everything)

### LINKED ISSUE #13746: Invalid tag name "extension 'node' property 'npmVersion'": Tags may not have any characters that encodeURIComponent encodes.
##### **Overview of the issue**

When I start command `gradlew npm_run_test -PnodeInstall --no-daemon` (from gitlab CI) I get the following error:

```
2 info using npm@6.14.8
3 info using node@v14.15.0
4 verbose npm-session 11f84155ce227e2d
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 timing stage:rollbackFailedOptional Completed in 1ms
8 timing stage:runTopLevelLifecycles Completed in 6ms
9 verbose stack Error: Invalid tag name "extension 'node' property 'npmVersion'": Tags may not have any characters that encodeURIComponent encodes.
9 verbose stack     at invalidTagName (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\npm-package-arg\npa.js:91:15)
9 verbose stack     at fromRegistry (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\npm-package-arg\npa.js:296:13)
9 verbose stack     at resolve (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\npm-package-arg\npa.js:81:12)
9 verbose stack     at npa (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\npm-package-arg\npa.js:52:10)
9 verbose stack     at C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\lib\install\deps.js:231:14
9 verbose stack     at C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\async-map.js:52:35
9 verbose stack     at Array.forEach (<anonymous>)
9 verbose stack     at C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\async-map.js:52:11
9 verbose stack     at Array.forEach (<anonymous>)
9 verbose stack     at asyncMap (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\async-map.js:51:8)
9 verbose stack     at exports.getAllMetadata (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\lib\install\deps.js:228:3)
9 verbose stack     at Installer.loadArgMetadata (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\lib\install.js:366:3)
9 verbose stack     at Installer.readGlobalPackageData (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\lib\install.js:667:8)
9 verbose stack     at Array.<anonymous> (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\bind-actor.js:15:8)
9 verbose stack     at LOOP (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\chain.js:15:14)
9 verbose stack     at chain (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\chain.js:20:5)
```
##### **Reproduce the error**

start command `gradlew npm_run_test -PnodeInstall --no-daemon`

##### **JHipster Version(s)**
7.0.0-beta.1

##### **JHipster configuration**
```
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "baseName": "new",
    "jhipsterVersion": "7.0.0-beta.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": true,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": [],
    "pages": [],
    "creationTimestamp": 1611734632808,
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "serviceDiscoveryType": "eureka",
    "reactive": false,
    "authenticationType": "oauth2",
    "packageName": "com.mycompany.myapp",
    "serverPort": "8080",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "gradle",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "vue",
    "withAdminUi": false,
    "clientTheme": "none",
    "enableTranslation": false,
    "nativeLanguage": "en",
    "packageFolder": "com/mycompany/myapp",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["en", "fr"]
  }
}

```

##### **Browsers and Operating System**

Windows 10
- [X] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by atomfrede**
This PR 

* updates gradle wrapper to 6.8.1
* remove no needed plugins (apt)
* update node plugin to rc7
* fix npm version definition for node plugin
* cleanup repositories

closes #13593
closes #13746

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/build.gradle.ejs [hunk 1]
```diff
@@ -18,7 +18,6 @@
 -%>
 buildscript {
     repositories {
-        mavenLocal()
         mavenCentral()
         gradlePluginPortal()
         maven { url "https://repo.spring.io/plugins-release" }
```
Comment (mraible) [inline generators/server/templates/build.gradle.ejs:23]: Is this line needed? It looks kinda funny to me.

File: generators/server/templates/gradle.properties.ejs [hunk 2]
```diff
@@ -21,7 +21,7 @@ profile=dev
 
 # Build properties
 nodeVersion=<%= NODE_VERSION %>
-npmVersion=<%= NPM_VERSION %>
+npmVersionProperty=<%= NPM_VERSION %>
```
Comment (mraible) [inline generators/server/templates/gradle.properties.ejs:24]: Why the change in name? Just curious.

File: generators/server/templates/build.gradle.ejs (hunk 1 above)
Comment (atomfrede) [reply generators/server/templates/build.gradle.ejs:23]: Need check. Maybe we can remove all additional buildscript repositories.

File: generators/server/templates/gradle.properties.ejs (hunk 2 above)
Comment (atomfrede) [reply generators/server/templates/gradle.properties.ejs:24]: Somehow the [property conflicts with the property name of the plugin](https://github.com/jhipster/generator-jhipster/issues/13746). As it is already the case for the old release version I suspect some gradle update or it never worked as expected (as we usually use npm directly instead maven/gradle in our ci).

So we could change the name or inline the versions in `build.gradle` as the properties are not used somewhere else.

File: generators/server/templates/build.gradle.ejs (hunk 1 above)
Comment (atomfrede) [reply generators/server/templates/build.gradle.ejs]: Removed all repositories except the gradle plugin portal.

**Comment by atomfrede** [comment]
Failing tests are unrelated, going to merge it.
